src/game/stats.c

```diff
@@ -0,0 +1,328 @@
+#include "game/stats.h"
+
+#include "game/draw.h"
+#include "game/gamebuf.h"
+#include "game/gameflow.h"
+#include "game/input.h"
+#include "game/output.h"
+#include "game/savegame.h"
+#include "game/screen.h"
+#include "game/shell.h"
+#include "game/text.h"
+#include "global/vars.h"
+#include "log.h"
+
+#include <stdio.h>
+#include <string.h>
+
+#define PIERRE_ITEMS 3
+#define SKATEKID_ITEMS 1
+#define COWBOY_ITEMS 1
+#define BALDY_ITEMS 1
+
+static int32_t m_CachedItemCount = 0;
+static FLOOR_INFO **m_CachedFloorArray = NULL;
+static int32_t m_LevelPickups = 0;
+static int32_t m_LevelKillables = 0;
+static int32_t m_LevelSecrets = 0;
+static bool m_KillableItems[MAX_ITEMS] = { 0 };
+static bool m_IfKillable[O_NUMBER_OF] = { 0 };
+
+int16_t m_PickupObjs[] = { O_PICKUP_ITEM1,   O_PICKUP_ITEM2,  O_KEY_ITEM1,
+                           O_KEY_ITEM2,      O_KEY_ITEM3,     O_KEY_ITEM4,
+                           O_PUZZLE_ITEM1,   O_PUZZLE_ITEM2,  O_PUZZLE_ITEM3,
+                           O_PUZZLE_ITEM4,   O_GUN_ITEM,      O_SHOTGUN_ITEM,
+                           O_MAGNUM_ITEM,    O_UZI_ITEM,      O_GUN_AMMO_ITEM,
+                           O_SG_AMMO_ITEM,   O_MAG_AMMO_ITEM, O_UZI_AMMO_ITEM,
+                           O_EXPLOSIVE_ITEM, O_MEDI_ITEM,     O_BIGMEDI_ITEM,
+                           O_SCION_ITEM,     O_SCION_ITEM2,   O_LEADBAR_ITEM,
+                           NO_ITEM };
+
+// Pierre has special trigger check
```

Comment:
So we don't need to check pods after all?

Reply:
Since we do the calculations before the objects are initialized, we check pods. This is useful because pods add mutants to the list of game items, but some pods (like the Abortion pod in Atlantis) don't have a trigger so there is no killable object associated with it.

Reply:
I remember some troll pods in the game that don't spawn anything, however reading the code for the pod reveals they always should spawn something. Do I remember wrong?

Reply:
Also Atlantis has a big pod that we should definitely not be counting.

Reply:
I checked all triggers and there are no pods in the base game that don't spawn anything. That only happens in UB but that's because they spawned pods with the Abortion flag when the abortion isn't in the level file hence the Lara torso bug.

The big pod won't be counted incorrectly because it doesn't have a trigger in Atlantis. I checked all the levels and the killables are correct.